tests/test_sentry.py

```diff
@@ -61,6 +161,172 @@ def test_subordinate(self, default_env, upload_scripts, wait_for_status):
         default_env.return_value = 'local'
         wait_for_status.return_value = mock_status
 
-        sentry = Talisman(['meteor', 'rsyslog-forwarder'])
+        sentry = Talisman(['meteor', 'rsyslog-forwarder'], timeout=self.timeout)
 
         self.assertTrue('rsyslog-forwarder/0' in sentry.unit)
+
+    @patch.object(Talisman, '__init__', Mock(return_value=None))
+    @patch('amulet.helpers.juju', Mock(return_value='status'))
+    @patch('amulet.waiter.status')
+    def test_wait_for_status(self, status):
+        status.return_value = mock_status
+        talisman = Talisman([], timeout=self.timeout)
+
+        talisman.wait_for_status('env', ['meteor'], self.timeout)
+        talisman.wait_for_status('env', ['old'], self.timeout)
+
+        self.assertRaises(TimeoutError, talisman.wait_for_status, 'env', ['pending'], self.timeout)
+        self.assertRaises(TimeoutError, talisman.wait_for_status, 'env', ['nopublic'], self.timeout)
+        self.assertRaisesRegexp(Exception, r'Error on unit.*hook failed',
+                                talisman.wait_for_status, 'env', ['errord'], self.timeout)
+        self.assertRaisesRegexp(Exception, r'Error on unit.*hook failed',
+                                talisman.wait_for_status, 'env', ['olderrord'], self.timeout)
+
+    @patch('amulet.helpers.juju', Mock(return_value='status'))
+    @patch('amulet.helpers.default_environment', Mock())
+    @patch.object(UnitSentry, 'upload_scripts', Mock())
+    @patch.object(UnitSentry, 'juju_agent')
+    @patch('amulet.waiter.status')
+    def test_wait(self, _status, juju_agent):
+        status = _status.return_value = deepcopy(mock_status)
+
+        def set_state(which, key, value):
+            status['services']['meteor']['units']['meteor/0'][which][key] = value
+
+        t = Talisman(['meteor'], timeout=self.timeout)
+        t.wait(self.timeout)
+
+        set_state('workload-status', 'current', 'unknown')
+        t.wait(self.timeout)
+
+        set_state('workload-status', 'current', 'blocked')
+        t.wait(self.timeout)
+
+        set_state('workload-status', 'current', 'active')
+        set_state('agent-status', 'current', 'executing')
+        self.assertRaises(TimeoutError, t.wait, self.timeout)
+
+        set_state('agent-status', 'current', 'idle')
+        set_state('agent-status', 'since', datetime.now().strftime('%d %b %Y %H:%M:%S'))
```

Comment:
Can we add a test for the 'idle' case where the 'since' timedelta > IDLE_THRESHOLD?